src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,321 @@
+                           generalCancerTypes.push(item);
+                       }
+                    });
+                    let clinicalGroupMatch: IClinicalGroupMatch = {
+                        trialAgeNumerical: age,
+                        trialOncotreePrimaryDiagnosis: {
+                            general: generalCancerTypes,
+                            not: notCancerTypes
+                        },
+                        matches: [],
+                        notMatches: []
+                    };
+                    const groupByGenomicAlteration = _.groupBy(groupByAge[age], 'genomicAlteration');
+                    _.forEach(groupByGenomicAlteration, (genomicAlterationGroup, genomicAlteration) => {
+                        const groupByPatientGenomic = _.groupBy(genomicAlterationGroup, 'patientGenomic');
+                        let genomicGroupMatch: IGenomicGroupMatch = {
+                            genomicAlteration: genomicAlteration,
+                            matches: []
+                        };
+                        _.forEach(groupByPatientGenomic, (patientGenomicGroup) => {
+                            let genomicMatch: IGenomicMatch = {
+                                trueHugoSymbol: patientGenomicGroup[0].trueHugoSymbol,
+                                trueProteinChange: patientGenomicGroup[0].trueProteinChange,
+                                sampleIds: []
+                            };
+                            const sampleIds = _.uniq(_.map(patientGenomicGroup, 'sampleId'));
+                            if (sampleIds.length > 1 && this.props.sampleManager) {
+                                this.props.sampleManager.samples.map((item:any) => {
+                                    if (sampleIds.includes(item.id)) {
+                                        genomicMatch.sampleIds.push(item.id);
+                                    }
+                                });
+                            } else {
+                                genomicMatch.sampleIds = sampleIds;
+                            }
+                            genomicGroupMatch.matches.push(genomicMatch);
+                        });
+                        if(genomicAlteration.includes('!')) {
+                            clinicalGroupMatch.notMatches.push(genomicGroupMatch);
+                        } else {
+                            clinicalGroupMatch.matches.push(genomicGroupMatch);
+                        }
+                    });
+                    matchedTrial['priority'] += setTrialMatchPriority(clinicalGroupMatch);
+                    armMatch.matches.push(clinicalGroupMatch);
+                });
+                matchedTrial['matches'].push(armMatch);
+            });
+            matchedTrials.push(matchedTrial);
+        });
+
+        return _.sortBy(matchedTrials, 'priority');
+    }
+
+    render() {
+        const detailedTrialMatches: Array<IDetailedTrialMatch> = this.buildDetailedTrialMatches();
```

Comment:
use IDetailedTrialMatch[] instead of Array<...>
You probably don't need to put type here as it should be inferred by typescript from return type of function..

Reply:
Changed to []